testutil/metrictest/assertions.go

```diff
@@ -0,0 +1,236 @@
+	})
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if ok {
+		return fmt.Errorf("expected not to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasOutboundHTTPRequestWithTLS checks there is a series matching:
+// request_total{
+//   direction="outbound",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9.]+",
+//   tls="true",
+//   dst_namespace="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   dst_serviceaccount="default"
+// }
+func HasOutboundHTTPRequestWithTLS(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction":          promm.Equals("outbound"),
+		"target_addr":        promm.Like(addrRE),
+		"target_ip":          promm.Like(ipRE),
+		"tls":                promm.Equals("true"),
+		"server_id":          promm.Equals(fmt.Sprintf("default.%s.serviceaccount.identity.linkerd.cluster.local", ns)),
+		"dst_namespace":      promm.Equals(ns),
+		"dst_serviceaccount": promm.Equals("default"),
+	},
+		promm.HasPositiveValue())
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("expected to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasOutboundHTTPRequestNoTLS checks there is a series matching:
+// request_total{
+//   direction="outbound",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9.]+",
+//   tls="no_identity",
+//   no_tls_reason="not_provided_by_service_discovery",
+//   dst_namespace="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   dst_serviceaccount="default"
+// }
+func HasOutboundHTTPRequestNoTLS(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction":          promm.Equals("outbound"),
+		"target_addr":        promm.Like(addrRE),
+		"target_ip":          promm.Like(ipRE),
+		"tls":                promm.Equals("no_identity"),
+		"no_tls_reason":      promm.Equals("not_provided_by_service_discovery"),
+		"dst_namespace":      promm.Equals(ns),
+		"dst_serviceaccount": promm.Equals("default"),
+	},
+		promm.HasPositiveValue())
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("expected to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasInboundSecureTCPTraffic checks there is a series matching:
+// tcp_open_total{
+//   direction="inbound",
+//   peer="src",
+//   tls="true",
+//   client_id="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   srv_kind="default",
+//   srv_name="all-unauthenticated",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9\.]+"
+// }
+func HasInboundSecureTCPTraffic(metrics, ns string) error {
```

Comment:
tioli:
 
```suggestion
func HasInboundTCPTrafficWithTLS(metrics, ns string) error {
```

Should be about the same length (+1 character). I think it's a bit awkward to say `HasInboundTLSTCPTraffic` so we could just put it at the end.

Reply:
Done: https://github.com/linkerd/linkerd2/pull/8547/commits/fca35c345a0aaca2384e8408b3af0564f1ec9cf5

TBH, if I get more variations (`WithXxxWithoutYyy`) I think I will parametrize these tests, just to avoid thinking about naming.